### PR TITLE
Disable config reload and throw a warning on receiving SIGHUP

### PR DESCRIPTION
**Pull request by seanchen1991**
Closes: #1885

## Description

Disables configuration reloading upon receiving a SIGHUP signal and throws a warning notifying the user of this fact and that the feature will be deprecated in the future. 

When a SIGHUP signal is sent to Hermes, the logs now look like this
```
2022-02-18T16:04:12.715032Z  INFO ThreadId(01) Hermes has started
2022-02-18T16:04:24.764642Z  WARN ThreadId(25) configuration reloading via SIGHUP has been disabled and will be deprecated in the future
```
______

### PR author checklist:
- [x] Added changelog entry, using [`unclog`](https://github.com/informalsystems/unclog).
- [ ] Added tests: integration (for Hermes) or unit/mock tests (for modules). 
- [x] Linked to GitHub issue.
- [x] Updated code comments and documentation (e.g., `docs/`).

### Reviewer checklist:

- [ ] Reviewed `Files changed` in the GitHub PR explorer.
- [ ] Manually tested (in case integration/unit/mock tests are absent).

### CODE REVIEW

File: relayer-cli/src/commands/start.rs [hunk 1]
```diff
@@ -59,9 +59,9 @@ impl Runnable for StartCmd {
 }
 
 /// Register the SIGHUP and SIGUSR1 signals, and notify the supervisor.
-/// - SIGHUP: Trigger a reload of the configuration.
+/// - [DEPRECATED] SIGHUP: Trigger a reload of the configuration.
 /// - SIGUSR1: Ask the supervisor to dump its state and print it to the console.
-fn register_signals(reload: ConfigReload, tx_cmd: Sender<SupervisorCmd>) -> Result<(), io::Error> {
+fn register_signals(_reload: ConfigReload, tx_cmd: Sender<SupervisorCmd>) -> Result<(), io::Error> {
```
Comment (adizere) [inline relayer-cli/src/commands/start.rs:64]: Should we also mark the `ConfigReload` struct or its `reload` method as `#[deprecated]` for extra caution around that module?

ref: https://doc.rust-lang.org/reference/attributes/diagnostics.html#the-deprecated-attribute

**Review by adizere — APPROVED** [review]
Pre-approving, thanks for the quick fix Sean!

Will let @romac have also a look since he knows the reloading module best.

File: relayer-cli/src/commands/start.rs (hunk 1 above)
Comment (romac) [reply relayer-cli/src/commands/start.rs:64]: I don't think it's needed since we don't expect anyone to rely on that programatically.

**Comment by romac** [comment]
LGTM modulo a nitpick. I'll take care of removing the config reload code in a follow-up PR.

**Review by romac — APPROVED** [review]

**Pull request merged**